test/jdk/java/io/File/libGetXSpace.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+#include <stdlib.h>
+#include "jni.h"
+#include "jni_util.h"
+#ifdef _WIN64
+#include <windows.h>
+#include <fileapi.h>
+#include <winerror.h>
+#else
+#include <sys/errno.h>
+#include <string.h>
+#if __APPLE__
+#include <sys/param.h>
+#include <sys/mount.h>
+#else
+#include <sys/statfs.h>
+#endif
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef _WIN64
+jboolean initialized = JNI_FALSE;
+BOOL(WINAPI * pfnGetDiskSpaceInformation)(LPCWSTR, LPVOID) = NULL;
+#endif
+
+//
+// root      the root of the volume
+// sizes[0]  total size:   number of bytes in the volume
+// sizes[1]  total space:  number of bytes visible to the caller
+// sizes[2]  free space:   number of free bytes in the volume
+// sizes[3]  usable space: number of bytes available to the caller
+//
+JNIEXPORT jboolean JNICALL
+Java_GetXSpace_getSpace0
+    (JNIEnv *env, jclass cls, jstring root, jlongArray sizes)
+{
+    jboolean totalSpaceIsEstimated = JNI_FALSE;
+    jlong array[4];
+    const jchar* chars = (*env)->GetStringChars(env, root, NULL);
+    if (chars == NULL) {
+        JNU_ThrowByNameWithLastError(env, "java/lang/RuntimeException",
+                                     "GetStringChars");
```

Comment:
Add a `return JNI_FALSE`; otherwise it falls through.

Reply:
Fixed in 9ae317e7b485d4ac56be25731165aafef694e9b5.